frontend/src/metabase/setup/components/Setup.jsx

```diff
@@ -56,9 +58,9 @@ export default class Setup extends Component {
     trackStructEvent("Setup", "Welcome");
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     this.setDefaultLanguage();
-    this.setDefaultDetails();
+    await this.setDefaultDetails();
```

Comment:
trackStepSeen relies on the current `props`. Could you move `await this.setDefaultDetails` below it?